code/modules/mob/living/simple_animal/hostile/roguetown/jitterskull.dm

```diff
@@ -0,0 +1,1338 @@
+        if(T.density)
+            continue
+        if(!isopenturf(T))
+            continue
+        options += T
+    if(!length(options))
+        return
+    // Prefer positions that still have line-of-sight to our reengagement target if possible
+    var/turf/choice
+    if(reengage_target && !QDELETED(reengage_target))
+        var/turf/rt = get_turf(reengage_target)
+        if(rt)
+            var/list/los_opts = list()
+            for(var/turf/opt in options)
+                if(has_line_of_sight(opt, rt))
+                    los_opts += opt
+            if(length(los_opts))
+                choice = pick(los_opts)
+    if(!choice)
+        choice = pick(options)
+    // Minimal flicker-style vanish/appear messaging
+    visible_message(span_notice("[src] flickers out of sight."))
+    if(QDELETED(src) || is_dying || QDELETED(choice))
+        return
+    forceMove(choice)
+    visible_message(span_notice("[src] flickers back into sight."))
+    // After teleporting, briefly translucent for a spooky effect, then restore quickly
+    alpha = 50
+    spawn(15)
+        if(!QDELETED(src))
+            alpha = original_alpha
+
+// Teleport near an atom at roughly a ring between min and max distance
+/mob/living/simple_animal/hostile/rogue/jitterskull/proc/teleport_near_atom(atom/A, min_range = 9, max_range = 11, require_los = TRUE)
+    if(is_dying || QDELETED(src) || QDELETED(A) || !A.loc)
+        return
+    var/turf/center = get_turf(A)
+    if(!center)
+        return
+    var/list/candidates = list()
+    var/minx = max(center.x - max_range, 1)
+    var/miny = max(center.y - max_range, 1)
+    var/maxx = center.x + max_range
+    var/maxy = center.y + max_range
+    var/z = center.z
+    for(var/turf/T in block(locate(minx, miny, z), locate(maxx, maxy, z)))
+        var/d = get_dist(center, T)
+        if(d < min_range || d > max_range)
+            continue
+        if(T.density || !isopenturf(T))
+            continue
+        // Optionally restrict to spots with approximate clear LOS to the target turf
+        if(!require_los || has_line_of_sight(T, center))
+            candidates += T
+    if(!length(candidates))
+        return
+    var/turf/spot = pick(candidates)
+    if(QDELETED(src) || is_dying || QDELETED(spot))
+        return
+    forceMove(spot)
+
+// Simple LOS: step from A to B along approximate straight path; fail on opaque/dense turfs
+/mob/living/simple_animal/hostile/rogue/jitterskull/proc/has_line_of_sight(turf/from_turf, turf/to_turf)
+    if(!from_turf || !to_turf)
+        return FALSE
+    if(from_turf.z != to_turf.z)
+        return FALSE
+    var/turf/cur = from_turf
+    var/safety = 0
+    while(cur && cur != to_turf && safety < 256)
+        safety++
+        var/dir_to = get_dir(cur, to_turf)
+        var/turf/next = get_step(cur, dir_to)
+        if(!next)
+            return FALSE
+        if(next.opacity || next.density)
+            return FALSE
+        cur = next
+    return TRUE
```

Comment:
should probably use the global `is_in_sight()` helper instead of writing a new one